01-resources-v2/branch-gen.yml

```diff
@@ -0,0 +1,39 @@
+resources:
+- name: atc
+  type: git
+  source: {uri: https://github.com/concourse/atc}
+  space: {branch: master}
```

Comment:
Thinking about build matrices, wouldn't this need to be a map of arrays (i.e.`map[string][]string` in go)? E.g.:

```
spaces:
  branch: [master, development]
```

Or, is it called `space` on a resource, and `spaces` on a task? If so, why?

Reply:
On a resource, you would define a single `space: ...` if you only ever care to deal with a single space everywhere by default (similar to today's behavior, only the space-related config would be under `space:`, not `source:`).

It's singular because at least when we were initially planning this it felt like having resources configure multiple spaces would be harder to keep track of vs. having each job define its spaces on a case-by-case basis.

Reply:
Yep, makes sense to me!